Tanks/Assets/Scripts/Shell/MineScript.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MineScript : MonoBehaviour
+{
+    public LayerMask m_TankMask;
+    public ParticleSystem m_ExplosionParticles;
+    public AudioSource m_ExplosionAudio;
+    public float m_ExplosionForce = 1000f;
+    public float m_MaxLifeTime = 30f;
+    public float m_ExplosionRadius = 5f;
+
+
+    private void Start()
+    {
+        Destroy(gameObject, m_MaxLifeTime);
+    }
+
+
+    private void OnTriggerEnter(Collider other)
+    {
+        // Find all the tanks in an area around the shell and damage them.
+        Collider[] colliders = Physics.OverlapSphere(transform.position, m_ExplosionRadius, m_TankMask);
+
+        for (int i = 0; i < colliders.Length; i++)
+        {
+            Rigidbody targetRigidbody = colliders[i].GetComponent<Rigidbody>();
+            if (!targetRigidbody)
+                continue;
+
+            targetRigidbody.AddExplosionForce(m_ExplosionForce, transform.position, m_ExplosionRadius);
+
+            TankHealth targetHealth = targetRigidbody.GetComponent<TankHealth>();
+
+            if (!targetHealth)
+                continue;
+
+            float damage = 30f;
+
+            targetHealth.TakeDamage(damage);
+
+            m_ExplosionParticles.transform.parent = null;
+
+            m_ExplosionParticles.Play();
+
+            m_ExplosionAudio.Play();
+
+            Destroy(m_ExplosionParticles.gameObject, m_ExplosionParticles.duration);
```

Comment:
Looks like ParticleSystem.duraction is deprecated and you should use main.duraction. Keep an eye out for green underlines in Visual Studio, especially when following older tutorials